onmt/utils/distributed.py

```diff
@@ -158,9 +158,10 @@ def signal_handler(self, signalnum, stackframe):
         raise Exception(msg)
 
 
-def batch_producer(generator_to_serve, queue, semaphore, opt):
+def batch_producer(generator_to_serve, queue, semaphore, opt, device_id):
     """Produce batches to `queues` from `generator_to_serve`."""
-    init_logger(opt.log_file)
+    if opt.verbose or device_id == 0:
```

Comment:
What happens if an error is raised in other processes?

Reply:
So maybe we set the log level to be error/warning for other processes?